### ISSUE TITLE
Add Tree of Life "Stance" to the Events that Trigger Outfit Swap

### ISSUE THREAD

**Issue by dbmamaz** [cdmichaelb/Outfitter#103]
I've been trying to get my healing set to automatically load when I go in to TOL form and it wont work.  I can get it to load on caster form, but not TOL.  I tried to uninstall and reboot but it kept my data somewhere.  i've deleted and rebuilt and i still cant get my TOL form to automate loading.  I was also having a few other issues - like before i deleted and rebuilt, it was loading my flight form outfit to my cat?  i'm sure i'm missing something here!

**Comment by DrSterni** [comment]
I have the exact same problem and tried to find some suggestions in the source code. Unfortunately I can't find any errors... What do the authors say?

**Comment by dbmamaz** [comment]
Havent heard anything yet but I really appreciate you chiming in!

**Comment by DrSterni** [comment]
Thanks, Cara. To be honest I am not very experienced in writing stuff with LUA-scripts. But anyways: I compared the scripts i.e. for the change of outfits when I enter the cat- or the flight-form. They work very well. Except the triggers I can't even see any differences at all. Why would the scripts work with the other forms, but not the "TREE"? I don't get it. Maybe the trigger itself is not correct?

**Comment by dbmamaz** [comment]
yeah thats what i wonder, if the trigger is messy somehow.  I've never played w LUA at all tho i've considered checking it out occasionally . . . . SQL is my jam tho

**Comment by dbmamaz** [comment]
Thinking old-school, i wonder if you could take a branch and add a statement to output to chat the trigger that comes from form changes and see what it actually outputs.  Currently my TOL trigger actually sort of seems to be working except the outfitter ui shows both my TOL outfit and flight form accessory as active  but doesnt actually equip the trinket for flight form trinket.  I'm starting to get used to just using the mini-map button every time i change form . . .

**Comment by Gogo1951** [comment]
@dbmamaz @DrSterni is this still an issue?

**Comment by DrSterni** [comment]
Sorry for my long-time absence. I will look it up tonight.

**Comment by dbmamaz** [comment]
Since there is now a crude (non-automated) outfit system in the base game in WOTLK, i have stopped using the add-on and I just manually change outfits - obviously less responsive when I change forms in the middle of a fight but more reliable.  And honestly sometimes the stats change because i went in and out of +mana or +stam gear was annoying.  

Thanks for checking in though!  I know these projects are all completely volunteer and I appreciate having it when I needed it!

**Comment by DrSterni** [comment]
Well, 1st of all: Yes, it's still an issue. Just looked it up a couple of minutes ago. Even with the new version (update from today) it's still not working. The change of gear while switching to cat-form or bear works very well. Changing to tree of life simply does not do anything.

@dbmamaz : Where can I find this ingame-feature? I could not find it at all... :-(. Is there a way to switch this on or off in order to make it work?

**Comment by dbmamaz** [comment]
Interface / Features - use equipment manager.  Its definitely not as intuitive as the add-on but its there.  again, no automatic changes.  Controlled from a button on the top left of the character screen 

I found comments saying it wasnt coming in prepatch, and then some saying it was, but could only find an article from 2007 or something saying how to turn it on

**Comment by DrSterni** [comment]
Thanks, I found it. But actually even if Outfitter does not work properly in this particular manner it is so much better than the ingame-tool that I still prefer it.

Why on earth the Outfitter doesn't change the gear only in the ToL-form just does not fit into my brain. Especially because the script seems to be exactly the same than with the other forms, only if course with the trigger "Tree of life".

I am using a german client, but even if I change the script to the trigger in german it does not work. There seems to be a mixup with the caster-form (human). I don't get it.

**Comment by dbmamaz** [comment]
are there actual spaces in the trigger for 'tree of life'?  might need something extra to wrap it? 

I agree teh add-on is much more convenient but i'm not ready to switch back to it - but partly because i'm not playing much and i was trying to figure out how to use raw log data to create powerbi charts lol

**Comment by Gogo1951** [comment]
Found the issue. (=

Should be an easy fix. Just need to set the Tree of Life up as an event for swapping gear. Thanks! 

Soon as I get to 80 I'll be working on making some updates.

**Comment by dbmamaz** [comment]
Curious is the tree-of-life different than the other forms, or did you have to set up all the forms the same way and you just missed it?

**Comment by DrSterni** [comment]
> Found the issue. (=
> Should be an easy fix. Just need to set the Tree of Life up as an event for swapping gear. Thanks!

Well, to be honest: This is exactly what we found out not to be working :-).

I compared a working script (Cat-Form) and the not working script (Tree-Form).

There are just 3 differences, the rest is absolutely identical. The 3 different lines are no. 1&2, 41 and 52:

The direct comparison looks like this:

Lines 1 & 2:
`-- $EVENTS CAT_FORM NOT_CAT_FORM OUTFIT_EQUIPPED`
`-- $DESC This outfit will be worn whenever you're in Cat Form`
versus
`-- $EVENTS TREE_FORM NOT_TREE_FORM OUTFIT_EQUIPPED`
`-- $DESC This outfit will be worn whenever you're in Tree Form`

Lines 41:
`elseif event == "CAT_FORM" then`
versus
`elseif event == "TREE_FORM" then`

Lines 52:
`elseif event == "NOT_CAT_FORM" then`
versus
`elseif event == "NOT_TREE_FORM" then`

This shows that there are only the triggers differing from each other, but they seem to be okay. Plus: The CAT_FORM works smooth as satin.

@Gogo1951 Is there anything you can do with this information?

**Comment by Gogo1951** [comment]
Hey @DrSterni , 

Wanna submit a PR? (=

But yeah I can use this when I get a minute. 

Got a few addons I'm working on now... but mostly just trying to level to 80 and fighting the queue boss. No way to log in and test / dev while I'm in an 8-hour queue. Sees like... bad design on Blizzard's part. =P

Groupie... https://www.curseforge.com/wow/addons/groupie

GDKPd... https://www.curseforge.com/wow/addons/gdkpdclassic

RankSentinel... https://www.curseforge.com/wow/addons/ranksentinel

**Comment by DrSterni** [comment]
I see. Thank you so much for taking the time to find the reason for this strange behaviour. I can absolutely understand the levelling situation, don't worry. Please take all the time you need. No need to hurry ;-)

As far as the PR (pull request, right? I am not too far into GitHub, to be honest) is concerned: If it helps I can do that, sure. I just wonder what this would do to your project. If this is sort of creating a fork esp. for this issue: I don't wanna ruin your great project :-)). If I would be able to program stuff in LUA, I would have done that already :-))

The other three projects you have been mentioning seem to be marvellous!! Especially "Groupie" is such a great idea!!
